File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/WritePropertyValuesResponse.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.eclipse.milo.opcua.stack.core.types.builtin.StatusCode;
import org.opcfoundation.webserver.types.common.UaChildId;

import java.util.HashMap;
import java.util.Map;

public class WritePropertyValuesResponse {
    private final Map<UaChildId, StatusCode> results;

    public WritePropertyValuesResponse()
    {
        results = new HashMap<>();
    }

    public void setWriteValueResult(String propertyName, StatusCode code)
    {
        results.put(new UaChildId(propertyName), code);
    }

    public Map<UaChildId, StatusCode> getResults() {
        return results;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/WritePropertyValuesResponse.ts
import { UaStatusCode } from "opcua-webapi-ts";
import { UaChildId } from "../../../types/common/UaChildId";

export class WritePropertyValuesResponse {
    private readonly _results: Map<string, UaStatusCode>;

    constructor() {
        this._results = new Map();
    }

    setWriteValueResult(propertyName: string | UaChildId, code: UaStatusCode): void {
        this._results.set(this.toKey(propertyName), code);
    }

    get results(): Map<string, UaStatusCode> {
        return this._results;
    }

    private toKey(propertyName: string | UaChildId): string {
        if (typeof propertyName === "string") {
            return propertyName;
        }

        return propertyName.toString();
    }
}